### PR TITLE
Adds peer code share document for relational rails. 

### PR DESCRIPTION
**Pull request by cjsim89**
… homepage.

# Backend Curriculum PR Template

### Description

Creating a peer code share template for students to complete with each other during Relational Rails evaluations. 

### Why are we making this update?

To help students explore and learn more about the topics covered throughout the first project in B2. 
  
### Type of update

- [ ] Minor update/fix -- no review requested
- [x] Moderate update -- review from Mod Team requested
- [ ] Major update -- review from Backend Team requested

### How will we measure the success of this change? 

Students will be able to see more examples of how they could write code in their first project, which can help solidify shaky topics before heading into the paired project. Also receiving and giving feedback to and from cohort colleagues. 

### What questions do you have/what do you want feedback on? (optional)

Questions and their order in the new peer_code_share document.

### CODE REVIEW

File: module2/projects/peer_code_share.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+title: Relational Rails - Peer Code Share
+---
+
+_[Back to Relational Rails Home](./relational_rails)_
+
+## Peer Code Share
+
+You will be partnered with another student to review each other's code and provide feedback.
+
+First, exchange Github repository links with the other student, and clone the repository. Set up the application locally by first checking that *their* database is named differently than *yours* - if you `drop` or re-seed a database of the exact same name as your project's, then yours will be affected too! To accomplish this, you could simply add the name of the student at the start of the database name, for example: `brian_relational_rails_development`, `brian_relational_rails_test`. You can find the database configuration in `config/database.yml`.
```
Comment (BrianZanti) [inline module2/projects/peer_code_share.md]: I think we can make this part optional. It should be okay if a students drops their development database for their project to view their partner's project. I can see some students getting hung up on changing the database names.

File: module2/projects/peer_code_share.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+Then, in your Terminal, run the following: 
+```ruby
+bundle install
+rails db:{drop,create,migrate,seed}
+rails s
+```
+
+Then, individually, take about 25 minutes to review the other person's code. You should write down answers to the following questions:
+
+1. Can you launch/use the application? Can the user stories be completed accurately? Do you find any bugs as you navigate through any pages?
```
Comment (BrianZanti) [inline module2/projects/peer_code_share.md]: I think we could use some more clarification on what it means to "complete" a user story. Something like "Can you perform the actions described in the user stories?"

File: module2/projects/peer_code_share.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+bundle install
+rails db:{drop,create,migrate,seed}
+rails s
+```
+
+Then, individually, take about 25 minutes to review the other person's code. You should write down answers to the following questions:
+
+1. Can you launch/use the application? Can the user stories be completed accurately? Do you find any bugs as you navigate through any pages? 
+1. Run their overall test suite (`bundle exec rspec`). Are there any failing tests? Are their tests strong/robust? (Strong tests utilize both obvious outcomes and edge cases.)
+1. Check their application's Models and the relationship to each other. Does the one-to-many relationship work? Are there any instance methods in their models? Are their models tested sufficiently (100%)?
```
Comment (BrianZanti) [inline module2/projects/peer_code_share.md]: Do we want them to just look at instance methods?

**Review by BrianZanti — COMMENTED** [review]
Hey Chris, thanks for putting up this PR! I like how you've structured this code share. I left a few comments on some things that could use clarification.

File: module2/projects/peer_code_share.md (hunk 1 above)
Comment (cjsim89) [reply module2/projects/peer_code_share.md]: Agreed - good point. I'll remove this part for now but leave the instructions for what to run in the terminal (which incidentally is the same steps for them to remake their database if needed anyway).

File: module2/projects/peer_code_share.md (hunk 2 above)
Comment (cjsim89) [reply module2/projects/peer_code_share.md]: Agreed - done

File: module2/projects/peer_code_share.md (hunk 3 above)
Comment (cjsim89) [reply module2/projects/peer_code_share.md]: I guess what I was getting at here was, if they have instance methods and they are tested (and those tests pass), then that's sufficient. Maybe in addition, check the model tests to be sure they are strong enough - adding more instances to the scenarios they already tested, for example, is something I ended up telling a lot of students in the past. 

To make this a little more effective, maybe it should also say "Are there any operations in a Controller or a View that should belong in this Model as methods instead?"

**Review by BrianZanti — APPROVED** [review]
